particula/util/size_distribution_convert.py

```diff
@@ -0,0 +1,173 @@
+"""Converts between different size distribution formats
+written with composition in mind and using a factory pattern
```

Comment:
@ngmahfouz What do you think about this direction, in terms of compostion patterns. I wanted to explore it on a smaller scale before particula core revision.

This replaces a case match function I was going to write to map all the conversions.